component/jsontypes.go

```diff
@@ -0,0 +1,28 @@
+package component
```

Comment:
Moved this to help separate flow.api and flow so they are no longer tightly coupled.

Reply:
I'd like for this change to be temporary, since it feels weird to me that type definitions used only for the API are moved to a high-level package like this. 

Do you have any long term ideas for how we can keep all API code isolated so that type definitions don't leak into shared packages? 

If we're looking for a temporary solution to work out circular dependencies, what do you think about putting these in a web/api/apitypes package?

Reply:
That sounds reasonable.